### ISSUE TITLE
Streamable UI slow to update error

### ISSUE THREAD

**Issue by GeoBrodas** [vercel/ai#2188]
### Description

- Head over to https://github.com/GeoBrodas/gemini-generative-ui
- Clone the repo
- Run `npm install`
- Add env `GOOGLE_VERTEX_KEY` (basically the API key for calling generative ui calls)
- Run `npm run dev`
- Visit `http://localhost:3000/gen-ui`
- Try 5-6 prompts, you''l be able to see the below console message

![image](https://github.com/vercel/ai/assets/68821192/b71021c6-7914-4ffa-b6f0-ebd7b8ff7dbc)

### Code example

Visit two main files to troubleshoot

`app\actions.tsx` and `app\gen-ui\page.tsx` to see the server call and client ui updation using `useChat`.

**Comment by GeoBrodas** [comment]
The error appears inspite of enign the streaming with `.done()`

**Comment by jeremyphilemon** [comment]
It looks like you haven't closed `uiStream` and `textStream` – I would add the following two lines right after this [line](https://github.com/GeoBrodas/gemini-generative-ui/blob/e0b93e5bc4fef86158a1b27ed27a09d4c832bcb6/app/actions.tsx#L210)

```ts
textStream.done();
uiStream.done();
```

with that being said, it looks like if a streamable value is not marked as done, the error message prints as `streamable ui is slow to update...` instead of `streamable value is slow to update...` – this will be fixed in #2190

**Comment by GeoBrodas** [comment]
Hi there! Thanks for the fix

The console errors have reduced but still comes when it executes the tool-call in this line

```js
          } else if (toolName === 'diagnose') {
            const { symptoms, country, state } = args;

            console.log('symptomsss', symptoms);
            console.log('location', country, 'from', state);

            uiStream.update(
              <div className="bg-yellow-200">You have holy water</div>
            );

            aiState.done({
              ...aiState.get(),
              interactions: [],
              messages: [
                ...aiState.get().messages,
                {
                  id: nanoid(),
                  role: 'assistant',
                  content: `Here's my diagnosis, you might be having `,
                  display: {
                    name: 'diagnose',
                    props: {
                      symptoms,
                    },
                  },
                },
              ],
            });
          }
        }
      }

      textStream.done();
      uiStream.done();
```

**Comment by GeoBrodas** [comment]
Sorry forgot to close `messageStream`. Thanks buddy again! Can close this issue now.

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **None**  
Confidence: **High**

### Rationale
The reported failure was ultimately a usage/support issue caused by unclosed streams, not broken SDK streaming behavior. The separate warning-text defect was handled elsewhere.

### Evidence
- The warning was caused by multiple streamable values not being closed with `.done()`.
- After closing `textStream`, `uiStream`, and the forgotten `messageStream`, the reporter said the issue could be closed.
- A maintainer noted a separate diagnostic wording bug (`streamable UI` instead of `streamable value`) and said it would be fixed in #2190.
